core/heartbeat.py

```
@@ -581,3 +609,146 @@ async def _behavior_create_dream(self) -> Optional[str]:
+                if t.completed_at and t.completed_at >= seven_days_ago
+            ]
+
+            if not recent_completed:
+                return None
+
+            # Check for daily streak
+            # Count consecutive days with at least one completion
+            streak_days = 0
+            check_day = time.time()
+
+            for day in range(7):
```

Ruff (0.14.14)

[warning] Loop control variable `day` not used within loop body

Rename unused `day` to `_day`

(B007)